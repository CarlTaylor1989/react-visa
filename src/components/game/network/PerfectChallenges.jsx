import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';

import './PerfectChallenges.scss';

export const PerfectChallenges = ({ current, total }) => (
  <div className="perfectChallenges">
    <Translate className="title" value="network.perfectChallenges" />
    <div className="completedContainer">
      <span className="icon" />
      <div className="completed">
        {current}
        <Translate value="network.perfectChallengesDivider" />
        {total}
      </div>
    </div>
  </div>
);

PerfectChallenges.propTypes = {
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired
};

export default PerfectChallenges;
