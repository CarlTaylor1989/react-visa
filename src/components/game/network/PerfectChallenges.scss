@import '../../../styles/mixins';
@import '../../../styles/variables';

$star-icon: '../../../styles/icons/star.svg';
$star-icon-height: 18px;
$star-icon-width: 18px;

.perfectChallenges {
  @include border-radius(16px 0 0 16px);
  background-color: $mini-dashboard-bg-colour;
  border: 1.5px solid $colour-white;
  border-right: 0;
  color: $colour-white;
  padding: 10px 20px 10px 25px;
  width: 130px;

  .title {
    display: block;
    font-size: 20px;
    font-weight: 800;
    margin-bottom: 5px;
    text-align: left;
  }

  .icon {
    background-image: url($star-icon);
    background-size: $star-icon-width $star-icon-height;
    height: $star-icon-height;
    margin-right: 10px;
    width: $star-icon-width;
  }

  .completed {
    font-size: 20px;
  }

  .completedContainer {
    @include align-items(center);
    display: flex;
    justify-content: flex-start;
  }
}
