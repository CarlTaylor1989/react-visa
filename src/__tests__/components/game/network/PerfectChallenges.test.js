/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import PerfectChallenges from '../../../../components/game/network/PerfectChallenges';

let mockProps;
beforeEach(() => {
  mockProps = {
    current: 22,
    total: 40
  };
});

test('renders the PerfectChallenges component', () => {
  const wrapper = shallow(<PerfectChallenges {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
