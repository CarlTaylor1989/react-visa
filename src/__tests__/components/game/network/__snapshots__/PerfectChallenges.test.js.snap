// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PerfectChallenges component 1`] = `
<div
  className="perfectChallenges"
>
  <Translate
    className="title"
    tag="span"
    value="network.perfectChallenges"
  />
  <div
    className="completedContainer"
  >
    <span
      className="icon"
    />
    <div
      className="completed"
    >
      22
      <Translate
        tag="span"
        value="network.perfectChallengesDivider"
      />
      40
    </div>
  </div>
</div>
`;
